Classes/git/PBGitRevSpecifier.m

```diff
@@ -14,26 +14,26 @@ @implementation PBGitRevSpecifier
 @synthesize parameters, description, workingDirectory;
 @synthesize isSimpleRef;
 
+// I believe this relates loosely to parts of git-check-ref-format.
+// cf. https://www.kernel.org/pub/software/scm/git/docs/git-check-ref-format.html
+//
+NS_INLINE BOOL ContainsComplexRefCharSequence(NSString *refString)
+{
+	return ([refString hasPrefix:@"-"] ||
+			[refString rangeOfCharacterFromSet:[NSCharacterSet characterSetWithCharactersInString:@" ~^:"]].location != NSNotFound ||
+			[refString rangeOfString:@".."].location != NSNotFound ||
+			[refString rangeOfString:@"@{"].location != NSNotFound);
+}
 
 // internal designated init
 - (id) initWithParameters:(NSArray *)params description:(NSString *)descrip
 {
-    self = [super init];
-    if (!self) return nil;
+	self = [super init];
+	if (!self) return nil;
+
 	parameters = params;
 	description = descrip;
-
-	if (([parameters count] > 1) || ([parameters count] == 0))
-		isSimpleRef =  NO;
-	else {
-		NSString *param = [parameters objectAtIndex:0];
-		if ([param hasPrefix:@"-"] ||
-			[param rangeOfCharacterFromSet:[NSCharacterSet characterSetWithCharactersInString:@"^@{}~:"]].location != NSNotFound ||
-			[param rangeOfString:@".."].location != NSNotFound)
-			isSimpleRef =  NO;
-		else
-			isSimpleRef =  YES;
-	}
+	isSimpleRef = (parameters.count == 1) && !ContainsComplexRefCharSequence(parameters[0]);
```

Comment:
This code is so confusing 😆. That's what I meant, but the more I look at it, the less I find it fool-proof.

Is it possible to have `parameters` be `nil` here ? Because otherwise `[nil count]` would return `nil` (via good ol' nil-messaging), and then you'd succeed on the 2nd part of the `OR` (`0 == nil`, making it a complex ref). If that's provably wrong, can you add an `NSParameterAssert`, that would make me less concerned... Null-annotating the header would be 👍 too, if it's not too crazy to do so.

I'm just wondering, but isn't `parameters[0]` (in your version) going to raise if the array is actually empty ?